tests/main/views/test_briefs.py

```diff
@@ -713,6 +714,88 @@ def test_list_briefs_no_pagination_if_user_id_supplied(self):
+            with freeze_time('2017-01-01 00:00:00'):
+                self.setup_dummy_briefs(1, lot='digital-outcomes', status=brief_status, brief_start=i + 1)
+            with freeze_time('2017-01-01 23:59:59'):
+                self.setup_dummy_briefs(1, lot='digital-outcomes', status=brief_status, brief_start=i + 6)
+
+            # The following briefs should be outside the time span
+            with freeze_time('2016-12-31 23:59:59'):
+                self.setup_dummy_briefs(1, lot='digital-outcomes', status=brief_status, brief_start=i + 11)
+            with freeze_time('2017-01-02 00:00:00'):
+                self.setup_dummy_briefs(1, lot='digital-outcomes', status=brief_status, brief_start=i + 16)
+
+        res = self.client.get('/briefs?{}=2017-01-01'.format(date_arg))
+        data = json.loads(res.get_data(as_text=True))
+
+        assert res.status_code == 200
+        assert len(data['briefs']) == 2
+
+    @pytest.mark.parametrize(
+        'date_arg,expected_count',
+        [('published_before', 5), ('withdrawn_before', 1), ('cancelled_before', 1), ('unsuccessful_before', 1)]
+    )
+    def test_list_briefs_filter_before_date(self, date_arg, expected_count):
+        for i, brief_status in enumerate(["draft", "withdrawn", "published", "cancelled", "unsuccessful"]):
+            with freeze_time('2016-12-31 23:59:59'):
+                self.setup_dummy_briefs(1, lot='digital-outcomes', status=brief_status, brief_start=i + 1)
+            # The following briefs should be filtered out ('before' is not inclusive)
+            with freeze_time('2017-01-01 00:00:00'):
+                self.setup_dummy_briefs(1, lot='digital-outcomes', status=brief_status, brief_start=i + 6)
+
+        res = self.client.get('/briefs?{}=2017-01-01'.format(date_arg))
+        data = json.loads(res.get_data(as_text=True))
+
+        assert res.status_code == 200
+        # When setting statuses on our dummy Briefs, we also set an 'old' published_at date,
+        #  so there will be more results for the `published_before` query than the others.
+        assert len(data['briefs']) == expected_count
+
+    @pytest.mark.parametrize(
+        'date_arg', ['published_after', 'withdrawn_after', 'cancelled_after', 'unsuccessful_after']
+    )
+    def test_list_briefs_filter_after_date(self, date_arg):
+        for i, brief_status in enumerate(["draft", "withdrawn", "published", "cancelled", "unsuccessful"]):
+            with freeze_time('2017-01-01 00:00:01'):
+                self.setup_dummy_briefs(1, lot='digital-outcomes', status=brief_status, brief_start=i + 1)
+            # The following briefs should be filtered out ('after' is not inclusive)
+            with freeze_time('2017-01-01 00:00:00'):
+                self.setup_dummy_briefs(1, lot='digital-outcomes', status=brief_status, brief_start=i + 6)
+
+        res = self.client.get('/briefs?{}=2017-01-01'.format(date_arg))
```

Comment:
This test feels weird. If you said to someone find me anything published after Thursday, you would either include both briefs published at 00:00 and 00:01 on Thursday because you are being inclusive of Thursday. Or you would include neither because you would only find briefs published on Friday.

Reply:
I'd prefer to include neither ('after' and 'before' imply non-inclusive to me). I'll change so that the `after_` case will hack the date in the same way as for the upper bound on `on_`, i.e. replace the param with 'param + 23:59:59'.